onnxruntime/core/providers/webnn/builders/impl/gemm_op_builder.cc

```diff
@@ -268,11 +268,43 @@ bool GemmOpBuilder::HasSupportedInputsImpl(const GraphViewer&, const Node& node,
     return false;
   }
 
-  if (op_type == "MatMulInteger") {
-    // The first decomposed op of MatMulInteger is DequantizeLinear, and so
-    // we only need to ensure it supports the input0_type.
+  if (op_type == "Gemm") {
+    return IsInputRankSupportedByOp(node, wnn_limits, logger) &&
+           IsDataTypeSupportedByOp(op_type, input0_type, wnn_limits, "a", "A", logger);
+  } else if (op_type == "MatMulInteger") {
+    // Check up to 4 inputs for MatMulInteger
+    for (size_t i = 0; i < std::min<size_t>(4, input_defs.size()); ++i) {
```

Comment:
`Input_defs.size()` will never be bigger than 4, (MatMulInteger has maximum input size 4), so you don't need to choose the min value.